src/library/basesqltablemodel.cpp

```diff
@@ -285,30 +251,22 @@ void BaseSqlTableModel::select() {
         qDebug() << "Rows actually received:" << rowInfo.size();
     }
 
-    // Adjust sort column to remove table columns and add 1 to add an id column.
-    int sortColumn = m_iSortColumn - m_tableColumns.size() + 1;
-
-    if (sortColumn < 0) {
-        sortColumn = 0;
-    }
-
     if (m_trackSource) {
-        // If we were sorting a table column, then secondary sort by id. TODO(rryan)
-        // we should look into being able to drop the secondary sort to save time
-        // but going for correctness first.
         m_trackSource->filterAndSort(trackIds, m_currentSearch,
                                      m_currentSearchFilter,
-                                     sortColumn, m_eSortOrder,
+                                     m_trackSourceOrderBy,
+                                     m_trackSourceSortColumn,
+                                     m_trackSourceSortOrder,
                                      &m_trackSortOrder);
 
         // Re-sort the track IDs since filterAndSort can change their order or mark
         // them for removal (by setting their row to -1).
         for (QVector<RowInfo>::iterator it = rowInfo.begin();
-             it != rowInfo.end(); ++it) {
-            // If the sort column is not a track column then we will sort only to
+                it != rowInfo.end(); ++it) {
```

Comment:
Both versions are correct. I like to keep this version, because the line break was hard to recognize in the original version because the / below.